app/models/dataset.rb

```diff
@@ -60,11 +60,27 @@ def publish!
       transaction do
         self.published!
         PublishingWorker.perform_async(self.id)
-        self.published_date = Time.now
+        self.last_published_at = Time.now
       end
     end
   end
 
+def update_legacy
```

Comment:
May be good to move those syncing methods to a service object. What do you think?